src/Akka.Hosting/AkkaHostingExtensions.cs

```diff
@@ -50,14 +47,21 @@ public static IServiceCollection AddAkka(this IServiceCollection services, strin
         /// Starts a background <see cref="IHostedService"/> that runs the <see cref="ActorSystem"/>
         /// and manages its lifecycle in accordance with Akka.NET best practices.
         /// </remarks>
-        public static IServiceCollection AddAkka(this IServiceCollection services, string actorSystemName, Action<AkkaConfigurationBuilder, ServiceProvider> builder)
+        public static IServiceCollection AddAkka(this IServiceCollection services, string actorSystemName, Action<AkkaConfigurationBuilder, IServiceProvider> builder)
         {
             var b = new AkkaConfigurationBuilder(services, actorSystemName);
-            var sp = services.BuildServiceProvider();
-            builder(b, sp);
+            services.AddSingleton<AkkaConfigurationBuilder>(sp =>
+            {
+                
+                builder(b, sp);
+                return b;
+            });
             
             // registers the hosted services and begins execution
-            b.Build();
+            b.Bind();
```

Comment:
Run the DI bindings exposed by Akka.NET outside the DI resolution scope so they're available as soon as the `IHost` starts.